Forbid renaming arguments of section variables (#14573)
coq/coq: doc/changelog/02-specification-language/14573-no-rename-var.rst

- **Removed:**

>   Argument renaming may not be used on section variables anymore (it was inconsistently applied previously)

Reviewer: ```suggestion
  Variables defined within sections may no longer be renamed with :cmd:`Argument` (this was previously applied inconsistently)
```

If I understand correctly

Reviewer: ```suggestion
  Arguments of section variables may no longer be renamed with :cmd:`Argument` (this was previously applied inconsistently)
```

It would be good to clarify this text too, if @SkySkimmer doesn't mind.  I won't lose sleep if it remains unchanged.
